test/features/parsoid/transform.js

```diff
@@ -15,197 +16,166 @@ describe('transform api', function() {
-            assert.contentType(res, contentTypes.wikitext);
+            assert.deepEqual(res.text, 'The modified HTML');
+            assert.match(res.headers['content-type'], mwUtil.constructRegex([contentTypes.wikitext]));
         });
     });
 
     it('html2wt, selser', () => {
-        return preq.post({
-            uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
-            body: {
-                html: testPage.html
-            }
-        })
+        return client.post(
+            `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
+            { html: testPage.html }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
-            assert.deepEqual(res.body, testPage.wikitext);
-            assert.contentType(res, contentTypes.wikitext);
+            assert.deepEqual(res.text, testPage.wikitext);
+            assert.match(res.headers['content-type'], mwUtil.constructRegex([contentTypes.wikitext]));
         });
     });
 
     it('html2wt with scrub_wikitext', () => {
-        return preq.post({
-            uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext`,
-            body: {
-                html: '<h2></h2>',
-                scrub_wikitext: 1
-            }
-        })
+        return client.post(
+            `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext`,
+            { html: '<h2></h2>', scrub_wikitext: 1 }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
-            assert.deepEqual(res.body, '');
+            assert.deepEqual(res.text, '');
         });
     });
 
     it('supports reversed order of properties in TimeUuid meta', () => {
         const newHtml = testPage.html.replace(/<meta property="mw:TimeUuid" content="([^"]+)"\/?>/,
             '<meta content="$1" property="mw:TimeUuid" />');
-        return preq.post({
-            uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
-            body: {
-                html: newHtml
-            }
-        })
+        return client.post(
+            `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
+            { html: newHtml }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
             const pattern = /Selser test/;
-            if (!pattern.test(res.body)) {
+            if (!pattern.test(res.text)) {
                 throw new Error(`Expected pattern in response: ${pattern
                 }\nSaw: ${JSON.stringify(res, null, 2)}`);
             }
-            assert.contentType(res, contentTypes.wikitext);
```

Comment:
This can be upstreamed to api-testing. Seems like a good general-purpose assertion.

Reply:
Done! @kevinbazira There is now an assert.contentType in the assert library: https://github.com/wikimedia/mediawiki-tools-api-testing/blob/master/lib/assert.js#L55

Reply:
Thanks @clarakosi and @Pchelolo, all tests are now using the `assert.contentTypes`.